src/django_otp/models.py

```diff
@@ -290,6 +309,35 @@ class VerifyNotAllowed:
     N_FAILED_ATTEMPTS = 'N_FAILED_ATTEMPTS'
 
 
+class GenerationThrottlingMixin(models.Model):
+    """
+    Mixin class for models that need throttling behaviour for generating
+    tokens.
+    """
+
+    def generate_is_allowed(self):
+        """
+        Checks if the time since the last token generation is greater than
+        configured (in seconds) by OTP_GENERATION_INTERVAL.
+        """
+        dt_now = timezone.now()
+        if (
+            not self.last_generated
+            or (dt_now - self.last_generated).total_seconds()
+            > settings.OTP_GENERATION_INTERVAL
```

Comment:
`ThrottlingMixin` has a `get_throttle_factor` method that subclasses must implement. Each device is responsible for managing its own setting. Default values are conventionally defined in relevant conf.py modules.

Reply:
Sounds good. I will add `get_generation_interval` method.

Reply:
> I will add `get_generation_interval` method.

Completed